internal/controller/grafanaorganization_controller.go

```diff
@@ -99,17 +101,79 @@ func (r GrafanaOrganizationReconciler) reconcileCreate(ctx context.Context, graf
 			return ctrl.Result{}, errors.WithStack(err)
 		}
 	}
-	_, err := grafanaAPI.Orgs.UpdateOrg(1, &grafanaAPIModels.UpdateOrgForm{
-		Name: "Shared Org.",
-	})
+
```

Comment:
I think checking the name is available first would be easier right? 

You use getByName and if it exists, you check that the org Id matches? If not you error and then you can do the create/update code?

Reply:
Because if you update orgA to orgB but orgB already exists then you're screwed

Reply:
If the status.orgId is empty then you create, otherwise you update

Reply:
I think it's kind of the same think, whether we first check the orgID or the name, in the end we'll have to execute the same checks